source/helpers.js

```diff
@@ -0,0 +1,47 @@
+const defineTypeWithFields = (definitons, BaseType, namespace, className, fields) => {
+  let classPath = className;
+  if (namespace !== '') classPath = `${namespace}.${className}`;
+  let SubType = BaseType.extend(classPath);
+  SubType.prototype.fields = function() {
+    return _.extend(BaseType.prototype.fields(), fields);
+  };
+  Space.resolvePath(namespace)[className] = SubType;
+  definitons[classPath] = SubType;
```

Comment:
Should this not be explicitly returned? I thought implicit returns were only in spec if the function body is a one-liner without brackets.